### PR TITLE
fix the problem that some config in profiles.default may not take effect

### LINKED ISSUE #5574: The config item in profiles.default section like dt_storage_blob_heavy_gc_valid_rate cannot take effect
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
config `blob_heavy_gc_valid_rate` to 0.7

### 2. What did you expect to see? (Required)
The config takes effect.

### 3. What did you see instead (Required)
The config doesn't take effect.

Note there are also some other config which have the same problem like `background_pool_size`.

### PR DESCRIPTION
**Pull request by lidezhu**
### What problem does this PR solve?

cherry-pick https://github.com/pingcap/tiflash/pull/5575

Issue Number: close #5574 

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by JaySon-Huang — APPROVED** [review]
lgtm

File: dbms/src/Storages/Page/ConfigSettings.cpp [hunk 1]
```diff
@@ -36,7 +36,6 @@ void mergeConfigFromSettings(const DB::Settings & settings, PageStorage::Config
 
     // V3 setting which export to global setting
     config.blob_heavy_gc_valid_rate = settings.dt_storage_blob_heavy_gc_valid_rate;
```
Comment (flowbehappy) [inline dbms/src/Storages/Page/ConfigSettings.cpp:38]: Also handle this line

File: dbms/src/Interpreters/Settings.h [hunk 2]
```diff
@@ -315,8 +315,7 @@ struct Settings
...
     M(SettingUInt64, dt_checksum_frame_size, DBMS_DEFAULT_BUFFER_SIZE, "Frame size for delta tree stable storage")                                                                                                                      \
                                                                                                                                                                                                                                         \
-    M(SettingDouble, dt_storage_blob_heavy_gc_valid_rate, 0.2, "Max valid rate of deciding a blob can be compact")                                                                                                                      \
-    M(SettingDouble, dt_storage_blob_block_alignment_bytes, 0, "Blob IO alignment size")                                                                                                                                                \
+    M(SettingDouble, dt_page_gc_threshold, 0.5, "Max valid rate of deciding a blob can be compact")                                                                                                                                     \
```
Comment (flowbehappy) [inline dbms/src/Interpreters/Settings.h:318]:
```suggestion
    M(SettingDouble, dt_page_gc_threshold, 0.5, "Max valid rate of deciding to do a GC in PageStorage")                                                                                                                                     \
```

File: dbms/src/Storages/Page/ConfigSettings.cpp (hunk 1 above)
Comment (lidezhu) [reply dbms/src/Storages/Page/ConfigSettings.cpp]: Fixed.

File: dbms/src/Interpreters/Settings.h (hunk 2 above)
Comment (lidezhu) [reply dbms/src/Interpreters/Settings.h]: Fixed.

File: dbms/src/Storages/Page/V3/tests/gtest_page_storage.cpp [hunk 3]
```diff
@@ -1635,5 +1636,28 @@ try
 }
 CATCH
 
+
+TEST_F(PageStorageTest, ReloadConfig)
+try
+{
+    auto & global_context = DB::tests::TiFlashTestEnv::getContext().getGlobalContext();
+    auto & settings = global_context.getSettingsRef();
+    auto old_dt_storage_blob_heavy_gc_valid_rate = settings.dt_storage_blob_heavy_gc_valid_rate;
```
Comment (flowbehappy) [inline dbms/src/Storages/Page/V3/tests/gtest_page_storage.cpp:1645]: I think we also need to change here @lidezhu

**Review by flowbehappy — APPROVED** [review]

**Pull request merged**